### ISSUE TITLE
Remove preprocessors step in boondoggle

### ISSUE THREAD

**Issue by cbosborn** [rigetti/quilc#358]
E.g. looking at the example script (now working thanks to #344)

```
(let ((*debug-noise-stream* *standard-output*)
      (chip-spec (quil::build-8q-chip)))
  (pipeline
   ((producer      ()        (make-instance 'producer-random
                                            :program-volume-limit 20
                                            :chip-specification chip-spec
                                            :respect-topology t))
    (preprocessors (i) (list (make-instance 'processor-identity)
                             (make-instance 'processor-quilc
                                            :executable-path "<path>")))
    (processors    (j) (list (make-instance 'processor-measures
                                            :chip-specification chip-spec)
                             (make-instance 'processor-identity)))
    (consumers     (j) (list (make-instance 'consumer-local-qvm
                                            :trials 1000)))
    (post-process  ()        (make-instance 'processor-L1-distance)))
   (stepA ()
          (produce-quil-program (producer)))
   (stepB ((i preprocessors))
          (progn
            (quil::print-parsed-program (stepA))
            (apply-process (preprocessors i) (stepA))))
   (stepC ((j processors) (i preprocessors))
          (apply-process (processors j) (stepB i)))
   (stepD ((j consumers) (i preprocessors))
          (consume-quil (consumers j) (stepC j i)))
   (stepE ((l consumers) (k preprocessors)
           (j consumers) (i preprocessors))
          (apply-process (post-process) (stepD j i) (stepD l k)))))
```

the purpose of `processors` is to apply measures _after_ compilation in the `preprocessors` step. Speaking with @ecpeterson this is an artifact of an older implementation of quilc, in which measure instructions were not acceptable in compilation.

The task here is to:

1. [ ] Move measure instruction addition into the random program producer step.
2. [ ] Perform compilation in the processors step, rather than preprocessors.
3. [ ] Remove the preprocessors step entirely.

cc @stylewarning, are there any other preprocesses we'd like to run that would justify keeping the preprocessors step around?

**Comment by stylewarning** [comment]
I don't see why it can't be removed entirely.

**Comment by ecpeterson** [comment]
Closed in #372.
